hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -76,6 +77,11 @@ public BaseCommitActionExecutor(HoodieEngineContext context, HoodieWriteConfig c
     // TODO : Remove this once we refactor and move out autoCommit method from here, since the TxnManager is held in {@link AbstractHoodieWriteClient}.
     this.txnManager = new TransactionManager(config, table.getMetaClient().getFs());
     this.lastCompletedTxn = TransactionUtils.getLastCompletedTxnInstantAndMetadata(table.getMetaClient());
+    // TODO: HUDI-2155 bulk insert support bucket index, HUDI-2156 cluster the table with bucket index.
+    if (this.config.getIndexType() == HoodieIndex.IndexType.BUCKET_INDEX
+        && (operationType == WriteOperationType.BULK_INSERT || operationType == WriteOperationType.CLUSTER)) {
```

Comment:
We might add new write operations and this check would let them through. e-g- Optimize introduced by
z-order PR

Reply:
Can we rewrite this based on what is actually Supported On bucket index

Reply:
will fix